raster-foundry/raster-foundry
# Project layer color correction UI

## Overview

This PR implements the project layer color correction UI.

**Relevant abstract links:**
- https://share.goabstract.com/48662bc6-7a12-43b3-881a-2fbe26e2b570
- https://share.goabstract.com/0c7e0f56-0b79-4c88-9e8d-c2119bcd21d4
- https://share.goabstract.com/d50057c7-e7cb-4ba1-89f9-b73812f99d58

### Checklist


### Demo

Color correct option under the `Color options` page:
![image](https://user-images.githubusercontent.com/2442245/53750454-d362ab00-3e77-11e9-97d1-3a36f2a901fe.png)

List of layer scenes to select:
![image](https://user-images.githubusercontent.com/2442245/53750499-eecdb600-3e77-11e9-9a69-40580ad7ceab.png)

List of layer scenes with one selected:
![image](https://user-images.githubusercontent.com/2442245/53750525-fdb46880-3e77-11e9-802c-d9a7de14fbe5.png)

Selected scene color correction (gamma collapsed):
![image](https://user-images.githubusercontent.com/2442245/53750613-3eac7d00-3e78-11e9-803f-f83f5ecba38d.png)

Selected scene color correction (gamma expanded):
![image](https://user-images.githubusercontent.com/2442245/53750639-5421a700-3e78-11e9-91b8-e02e1fa1752f.png)


### Notes

I'm a bit concerned that something here kills backsplash. It seems a bit random, but it's happened numerous times.

## Testing Instructions

 * Create some layers with scenes 
 * Try color correcting
 * See that you can reset the corrections from both the scene list (for all scenes in the layer) and the correction page (which will reset only the selected scenes).
 * See that changes are persisted
 * See if this thing kills backsplash

Closes #4704
